opgee/processes/compressor.py

```diff
@@ -53,59 +63,32 @@ def get_compressor_work_temp(field, inlet_temp, inlet_press, gas_stream, compres
         return work_sum, inlet_temp, inlet_press
 
     @staticmethod
-    def get_compression_ratio_stages(overall_compression_ratio_stages):
-        max_stages = len(_power)
-        compression_ratio_per_stages = []
-
-        for compression_ratio in overall_compression_ratio_stages:
-            for pow in _power:
-                if compression_ratio ** pow < max_stages:
-                    compression_ratio_per_stages.append(compression_ratio ** pow)
-                    break
-
-        return compression_ratio_per_stages
+    def get_compression_ratio_stages(
+        overall_compression_ratio_stages: Sequence[Q_Float]
+    ) -> Sequence[Q_IntTuple]:
+        compression_ratios: map[Optional[Q_IntTuple]] = map(
+            Compressor.get_compression_ratio_and_stage, overall_compression_ratio_stages
+        )
+        return [ratio for ratio in compression_ratios if ratio is not None]
 
     @staticmethod
-    def get_compression_ratio(overall_compression_ratio):
+    @ureg.wraps(("frac", None), "frac", strict=False)
+    def get_compression_ratio_and_stage(overall_compression_ratio: float) -> Optional[Q_IntTuple]:
         max_stages = len(_power)
-        result = 0
-
-        for pow in _power:
-            if overall_compression_ratio ** pow < max_stages:
-                result = overall_compression_ratio ** pow
-                return result
-
-    @staticmethod
-    def get_num_of_compression_stages(overall_compression_ratio_stages):
-        num_of_compression_stages = []
-        compression_ratio_per_stages = Compressor.get_compression_ratio_stages(overall_compression_ratio_stages)
-
-        for overall_compression_ratio, compression_ratio in \
-                zip(overall_compression_ratio_stages, compression_ratio_per_stages):
-            for pow in _power:
-                if overall_compression_ratio ** pow == compression_ratio:
-                    num_of_compression_stages.append(int(1 / pow))
-                    break
-
-        return num_of_compression_stages
-
-    @staticmethod
-    def get_num_of_compression(overall_compression_ratio):
-        result = 0
-        compression_ratio = Compressor.get_compression_ratio(overall_compression_ratio)
-
         for pow in _power:
-            if overall_compression_ratio ** pow == compression_ratio:
-                result = int(1 / pow)
-                return result
+            if overall_compression_ratio**pow < max_stages:
```

Comment:
I'd prefer to avoid the redundant calculation of `overall_compression_ratio ** pow`. My usual approach is to store it in a variable instead of computing twice. Granted, it's a small optimization, but the func is called in a loop.

Reply:
Happy to add that.